source/opt/ccp_pass.cpp

```diff
@@ -87,23 +107,30 @@ SSAPropagator::PropStatus CCPPass::VisitAssignment(ir::Instruction* instr) {
     if (it != values_.end()) {
       values_[instr->result_id()] = it->second;
       return SSAPropagator::kInteresting;
+    } else if (IsVaryingValue(it->second)) {
```

Comment:
This looks odd.  If `it == values_.end()` then load `it->second`?

Reply:
Fixed, thanks.  The case can actually never happen because the propagator is guaranteed to always have generated a value of the RHS of an OpCopyObject (it works in def-use and dominator order, after all).  However, it would've failed to mark the LHS varying if the RHS was varying and would mark the instruction not-interesting, causing an additional go around the propagator.